File: app/local_plugins/vendors/llm_openai/android/src/main/kotlin/com/aiagent/llm_openai/LlmOpenaiPlugin.kt
package com.aiagent.llm_openai

import android.util.Log
import com.aiagent.plugin_interface.NativeServiceRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * LlmOpenaiPlugin — OpenAI-compatible LLM Flutter 插件
 *
 * 主要角色：在 onAttachedToEngine 时注册 NativeLlmService 到 NativeServiceRegistry，
 * 供 Agent 类型插件（agent_chat 等）在原生层直接调用。
 *
 * Dart 侧的纯 HTTP 实现 (llm_openai_plugin.dart) 保留向后兼容。
 */
class LlmOpenaiPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // 注册 OpenAI 兼容的 LLM 厂商（共用一套 HTTP SSE 协议）。
        // 火山引擎（volcengine）已拆到 llm_volcengine 插件独立实现，不在此列表。
        val vendors = listOf("openai", "deepseek", "moonshot", "zhipu", "qwen", "minimax", "baichuan")
        for (vendor in vendors) {
            NativeServiceRegistry.registerLlm(vendor) { LlmOpenaiService() }
        }
        Log.d("LlmOpenaiPlugin", "Registered NativeLlmService for vendors: $vendors")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // no-op
    }
}

File: apps/client/local_plugins/vendors/llm_openai/ios/Classes/LlmOpenaiPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// OpenAI-compatible LLM Flutter plugin entry.
///
/// At registration time we register `LlmOpenaiService` against every vendor
/// id that speaks the OpenAI chat-completions SSE protocol. The volcengine
/// Ark API is split out into its own plugin (`llm_volcengine`).
public class LlmOpenaiPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.llm_openai", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        let vendors = ["openai", "deepseek", "moonshot", "zhipu", "qwen", "minimax", "baichuan"]
        for vendor in vendors {
            NativeServiceRegistry.shared.registerLlm(vendor) { LlmOpenaiService() }
        }
        os_log("Registered NativeLlmService for vendors: %{public}@",
               log: log, type: .debug, vendors.description)
    }
}
